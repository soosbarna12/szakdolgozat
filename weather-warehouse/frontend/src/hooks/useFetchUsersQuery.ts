import { useQuery } from "@tanstack/react-query";
import { User } from "../types/userData.type";
import axios from "../utils/axiosConfig";


export function useFetchUsersQuery() {
  
  const { data, error, isLoading, refetch } = useQuery<User[]>({
    queryKey: ["users"],
    queryFn: async () => {
      const response = await axios.get(`/api//user/userData`);

      return response.data as User[];
    },
  });

  return { data, error, isLoading, refetch };
}
