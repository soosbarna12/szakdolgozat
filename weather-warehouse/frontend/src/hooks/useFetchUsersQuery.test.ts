import '@testing-library/jest-dom';
import { waitFor } from '@testing-library/react';
import axios from '../utils/axiosConfig';
import { renderHookWithQueryClient } from '../utils/test/renderHookWithQueryClient';
import { useFetchUsersQuery } from './useFetchUsersQuery';

jest.mock('../utils/axiosConfig');

describe('useFetchUsersQuery', () => {
  const mockUsers = [
    { id: 1, name: 'John Doe', email: 'john.doe@example.com' },
    { id: 2, name: 'Jane Smith', email: 'jane.smith@example.com' },
  ];

  afterEach(() => {
    jest.clearAllMocks();
    (axios.get as jest.Mock).mockReset();
  });

  beforeEach(() => {
  });

  it('matches useFetchUsersQuery hook snapshot', async () => {
    (axios.get as jest.Mock).mockReturnValue({ data: mockUsers });
    const { result } = renderHookWithQueryClient(() => useFetchUsersQuery());

    await waitFor(() => {
      expect(result.current.data).not.toBeUndefined();
    });

    expect(result.current).toMatchSnapshot();
  });

  it('fetches users successfully', async () => {
    (axios.get as jest.Mock).mockReturnValue({ data: mockUsers });

    const { result } = renderHookWithQueryClient(() => useFetchUsersQuery());

    await waitFor(() => {
      expect(result.current.data).toEqual(mockUsers);
      expect(result.current.isLoading).toBe(false);
    });
  });

  it('allows refetching users', async () => {
    (axios.get as jest.Mock).mockReturnValue({ data: mockUsers });

    const { result } = renderHookWithQueryClient(() => useFetchUsersQuery());

    await waitFor(() => {
      expect(result.current.data).toEqual(mockUsers);
    });

    (axios.get as jest.Mock).mockReturnValue({ data: [] });

    await result.current.refetch();

    await waitFor(() => {
      expect(result.current.data).toEqual([]);
    });
  });
});