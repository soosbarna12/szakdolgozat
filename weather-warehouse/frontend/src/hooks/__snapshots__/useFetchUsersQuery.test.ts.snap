// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useFetchUsersQuery matches useFetchUsersQuery hook snapshot 1`] = `
{
  "data": [
    {
      "email": "john.doe@example.com",
      "id": 1,
      "name": "John Doe",
    },
    {
      "email": "jane.smith@example.com",
      "id": 2,
      "name": "Jane Smith",
    },
  ],
  "error": null,
  "isLoading": false,
  "refetch": [Function],
}
`;
